### ISSUE TITLE
Move logic more away from logic and into `@custom-element`

### ISSUE THREAD

**Issue by emolr** [framejs/framejs#8]
Just a thought! 

As the framework grows, it becomes a bit more complex to remember what goes where. 

At the moment logic is at most placed in the decorator functions themself. These decisions where made in order to make the library more lightweight, the lesser decorator functions where used. 

Now having built numerous components it feels a bit like overhead to define for example 20 x `@Attribute()` for mimicking a full component like `<input>` and having the functionality for the attribute located in `@customElement`would mean that a more React like approach could be used. 

for example now it looks like:
```
import { CustomElement, Attribute } from '@framejs/core';

@customElement({tag: 'my-element'})
export class MyElement extends HTMLElement {
    @Attribute() checked: boolean = false;
}
```

But changing the architecture a bit it could also support this:
```
import { CustomElement } from '@framejs/core';

@customElement({tag: 'my-element'})
export class MyElement extends HTMLElement {
  static get props() {
    return {
       checked: false
    }
  }

  static get propTypes() {
    return {
       checked: Boolean
    }
  }

  static get attributeReflectedProps() {
    return [checked]
  }
}
```

In short simple components it doesn't seem beneficial, but in large components it could be, and could down the road mean that framejs would work without typescript.
